.github/workflows/lint-go.yaml

```diff
@@ -57,6 +57,24 @@ jobs:
           skip-cache: true
           args: "--out-${NO_FUTURE}format colored-line-number --verbose --modules-download-mode=vendor"
 
+  staticcheck:
+    name: Lint source code using staticcheck (to detect duplicate package imports)
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@93397bea11091df50f3d7e59dc26a7711a8bcfbe # v4.1.0
+        with:
+          # renovate: datasource=golang-version depName=go
+          go-version: 1.21.0
+      - name: Checkout code
+        uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3.6.0
+        with:
+          persist-credentials: false
+      - name: Run staticcheck linting
```

Comment:
Static lint can be run as part of golangci-lint, do we need to install it separately or can we just enable that lint in our golangci-lint.yaml?

Reply:
Hey there. I am working on this.

I removed the `.staticcheck.conf` file and in the `.golangci.yaml` file, I made this change -
```yaml
staticcheck:
    go: "1.21.0"
    checks: ["ST1019"] # Check for duplicate package imports  
```

To test, whether it works as intended or not, I intentionally created a duplicate package import in the `pkg/ebpf/map_register.go` file. Then I ran `golangci-lint run  --config=./.golangci.yaml ./pkg/ebpf`. But surprisingly no errors were thrown.

Reply:
NB: golangci-lint staticcheck is not the same as the binary. They just copied some rules: https://golangci-lint.run/usage/linters/#staticcheck

Reply:
Ahhh. Why would they reuse the name like that 🤦 . Okay well then I guess this thread can be resolved.